azkaban-common/src/main/java/azkaban/executor/AbstractExecutorManagerAdapter.java

```diff
@@ -179,12 +195,45 @@ public int getExecutableFlows(final int projectId, final String flowId, final in
     return this.executorLoader.doRampActions(rampActions);
   }
 
-  protected String uploadExecutableFlow(
-      final ExecutableFlow exflow, final String userId, final String flowId,
+  @Override
+  public Status getStartStatus() {
+    return Status.PREPARING;
```

Comment:
This is default behavior. For existing dispatch methods, the start state for flow is PREPARING. But it can be overridden for any implementation for e.g. for containerization, it is READY.